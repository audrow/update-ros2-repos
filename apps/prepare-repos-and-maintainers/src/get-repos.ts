import fetch from "node-fetch"
import * as z from 'zod'

const repoFromUrlSchema = z.object({
  org: z.string(),
  name: z.string(),
  url: z.string().url(),
  validUrl: z.boolean().nullable(),
})

type RepoFromUrl = z.infer<typeof repoFromUrlSchema>

const repoSchema = z.object({
  maintainers: z.array(z.string().min(1)),
}).merge(repoFromUrlSchema)

export type Repo = z.infer<typeof repoSchema>

export async function getRepos({
  maintainersCsvData,
  headingRowNumber,
  maintainersStartColumnLetter,
  reposColumnLetter,
  githubUrlPrefix,
  isCheckUrls,
}: {
  maintainersCsvData: string[][]
  headingRowNumber: number
  maintainersStartColumnLetter: string
  reposColumnLetter: string
  githubUrlPrefix: string
  isCheckUrls: boolean
}) {
  const heading = maintainersCsvData[headingRowNumber - 1]
  const people = z
    .array(z.string().min(2))
    .parse(heading.slice(letterToIndex(maintainersStartColumnLetter)))

  const repos = await Promise.all(
    maintainersCsvData
      .splice(headingRowNumber)
      .filter((row: string[]) => {
        const repoString = row[letterToIndex(reposColumnLetter)]
        return repoString.split('/').length === 2
      })
      .map(async (row: string[]) => {
        const repoString = z
          .string()
          .parse(row[letterToIndex(reposColumnLetter)])
        const isMaintainer = z
          .array(
            z.preprocess((x: unknown) => {
              if (x === '') {
                return false
              }
              return Number.parseFloat(x as string) > 0
            }, z.boolean()),
          )
          .length(people.length)
          .parse(row.slice(letterToIndex(maintainersStartColumnLetter)))

        const maintainers = people.filter(
          (_: string, index: number) => isMaintainer[index],
        )

        const repoFromUrl = await getRepoFromString(
          repoString,
          githubUrlPrefix,
          isCheckUrls,
        )

        return repoSchema.parse({
          ...repoFromUrl,
          maintainers,
        })
      }),
  )

  const maintainers = repos.reduce((acc: string[], repo) => {
    repo.maintainers.forEach((maintainer) => {
      if (!acc.includes(maintainer)) {
        acc.push(maintainer)
      }
    })
    return acc
  }, [])

  return {
    maintainers,
    repos,
  }
}

export async function getRepoFromString(
  repoString: string,
  githubUrlPrefix: string,
  isCheckUrl = false,
): Promise<RepoFromUrl> {
  const [org, name] = repoString.split('/')
  const url = `${githubUrlPrefix}${org}/${name}`

  let validUrl: boolean | null = null
  if (isCheckUrl) {
    const {status} = await fetch(url)
    validUrl = status === 200
    if (validUrl === false) {
      console.log(`repo ${url} returned status ${status}`)
    }
  }

  return {
    org,
    name,
    url,
    validUrl,
  }
}

function letterToIndex(letter: string): number {
  return z.string().length(1).parse(letter).charCodeAt(0) - 97
}
