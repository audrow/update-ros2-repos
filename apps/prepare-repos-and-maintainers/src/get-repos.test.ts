import {parse as csvParse} from 'csv-parse/sync'
import fs from 'fs'
import path from 'path'
import {getRepoFromString, getRepos} from './get-repos'

jest.setTimeout(10000)

describe('getRepos', () => {
  it('should return repos', async () => {
    const maintainersCsv = fs.readFileSync(
      path.join(__dirname, '..', 'data', '2022-09-14-maintainers.csv'),
    )
    const reposColumnLetter = 'a'
    const maintainersStartColumnLetter = 'n'
    const headingRowNumber = 2
    const isCheckUrls = false
    const githubUrlPrefix = 'https://github.com/'
    const data = csvParse(maintainersCsv)
    const {repos, maintainers} = await getRepos({
      maintainersCsvData: data,
      headingRowNumber,
      maintainersStartColumnLetter,
      reposColumnLetter,
      githubUrlPrefix,
      isCheckUrls,
    })
    expect(repos).toMatchSnapshot()
    expect(maintainers).toMatchSnapshot()
  })
})

describe('getRepoFromString', () => {
  it('gets a repo that exists', async () => {
    const repoString = 'ros2/rclcpp'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = true
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "rclcpp",
        "org": "ros2",
        "url": "https://github.com/ros2/rclcpp",
        "validUrl": true,
      }
    `)
  })
  it("can't find url for a repo that doesn't exist", async () => {
    const repoString = 'ros2/not-a-repo'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = true
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "not-a-repo",
        "org": "ros2",
        "url": "https://github.com/ros2/not-a-repo",
        "validUrl": false,
      }
    `)
  })
  it('can skip checking a URL', async () => {
    const repoString = 'ros2/rclcpp'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = false
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "rclcpp",
        "org": "ros2",
        "url": "https://github.com/ros2/rclcpp",
        "validUrl": null,
      }
    `)
  })
})
