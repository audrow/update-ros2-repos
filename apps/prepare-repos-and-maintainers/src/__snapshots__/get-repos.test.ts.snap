// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRepos should return repos 1`] = `
[
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "hidmic",
    ],
    "name": "rclcpp",
    "org": "ros2",
    "url": "https://github.com/ros2/rclcpp",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
      "adityapande-1995",
    ],
    "name": "rclpy",
    "org": "ros2",
    "url": "https://github.com/ros2/rclpy",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "codebot",
    ],
    "name": "ros2",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "rviz",
    "org": "ros2",
    "url": "https://github.com/ros2/rviz",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "audrow",
    ],
    "name": "ros2_documentation",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2_documentation",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "adityapande-1995",
      "methylDragon",
    ],
    "name": "launch",
    "org": "ros2",
    "url": "https://github.com/ros2/launch",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
      "audrow",
    ],
    "name": "ros2cli",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2cli",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "geometry2",
    "org": "ros2",
    "url": "https://github.com/ros2/geometry2",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
      "adityapande-1995",
      "quarkytale",
      "methylDragon",
    ],
    "name": "rosidl",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ivanpauno",
    ],
    "name": "rmw_cyclonedds",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_cyclonedds",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ament_lint",
    "org": "ament",
    "url": "https://github.com/ament/ament_lint",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "audrow",
    ],
    "name": "rcl",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
      "gbiggs",
    ],
    "name": "rmw_fastrtps",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_fastrtps",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "cottsay",
      "nuclearsandwich",
    ],
    "name": "ci",
    "org": "ros2",
    "url": "https://github.com/ros2/ci",
    "validUrl": null,
  },
  {
    "maintainers": [
      "audrow",
    ],
    "name": "docker_images",
    "org": "osrf",
    "url": "https://github.com/osrf/docker_images",
    "validUrl": null,
  },
  {
    "maintainers": [
      "adityapande-1995",
      "methylDragon",
    ],
    "name": "launch_ros",
    "org": "ros2",
    "url": "https://github.com/ros2/launch_ros",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "ros_buildfarm_config",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_buildfarm_config",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
      "adityapande-1995",
      "audrow",
    ],
    "name": "demos",
    "org": "ros2",
    "url": "https://github.com/ros2/demos",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
      "methylDragon",
    ],
    "name": "image_common",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/image_common",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
      "quarkytale",
      "methylDragon",
    ],
    "name": "ros1_bridge",
    "org": "ros2",
    "url": "https://github.com/ros2/ros1_bridge",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ament_cmake",
    "org": "ament",
    "url": "https://github.com/ament/ament_cmake",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "rep",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rep",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
      "ahcorde",
    ],
    "name": "rcutils",
    "org": "ros2",
    "url": "https://github.com/ros2/rcutils",
    "validUrl": null,
  },
  {
    "maintainers": [
      "tfoote",
    ],
    "name": "common_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/common_interfaces",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
      "quarkytale",
    ],
    "name": "rosidl_python",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_python",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "robot_state_publisher",
    "org": "ros",
    "url": "https://github.com/ros/robot_state_publisher",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "methylDragon",
    ],
    "name": "rmw",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
      "adityapande-1995",
    ],
    "name": "examples",
    "org": "ros2",
    "url": "https://github.com/ros2/examples",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "message_filters",
    "org": "ros2",
    "url": "https://github.com/ros2/message_filters",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "system_tests",
    "org": "ros2",
    "url": "https://github.com/ros2/system_tests",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_bag",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_bag",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "urdfdom",
    "org": "ros",
    "url": "https://github.com/ros/urdfdom",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "pluginlib",
    "org": "ros",
    "url": "https://github.com/ros/pluginlib",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "rqt_image_view",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_image_view",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
    ],
    "name": "design",
    "org": "ros2",
    "url": "https://github.com/ros2/design",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rcl_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl_interfaces",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "methylDragon",
    ],
    "name": "rcpputils",
    "org": "ros2",
    "url": "https://github.com/ros2/rcpputils",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "yaml_cpp_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/yaml_cpp_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "class_loader",
    "org": "ros",
    "url": "https://github.com/ros/class_loader",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "qt_gui_core",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/qt_gui_core",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "orocos_kdl_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/orocos_kdl_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "rmw_implementation",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_implementation",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "rosdoc2",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosdoc2",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "joint_state_publisher",
    "org": "ros",
    "url": "https://github.com/ros/joint_state_publisher",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "kdl_parser",
    "org": "ros",
    "url": "https://github.com/ros/kdl_parser",
    "validUrl": null,
  },
  {
    "maintainers": [],
    "name": "ros_tutorials",
    "org": "ros",
    "url": "https://github.com/ros/ros_tutorials",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_graph",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_graph",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_reconfigure",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_reconfigure",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "osrf_pycommon",
    "org": "osrf",
    "url": "https://github.com/osrf/osrf_pycommon",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "buildfarm_perf_tests",
    "org": "ros2",
    "url": "https://github.com/ros2/buildfarm_perf_tests",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "libyaml_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/libyaml_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
    ],
    "name": "rcl_logging",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl_logging",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "realtime_support",
    "org": "ros2",
    "url": "https://github.com/ros2/realtime_support",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
      "quarkytale",
    ],
    "name": "rosidl_typesupport",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_typesupport_fastrtps",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport_fastrtps",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rosindex",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosindex",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "angles",
    "org": "ros",
    "url": "https://github.com/ros/angles",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "bond_core",
    "org": "ros",
    "url": "https://github.com/ros/bond_core",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "geometry_tutorials",
    "org": "ros",
    "url": "https://github.com/ros/geometry_tutorials",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "resource_retriever",
    "org": "ros",
    "url": "https://github.com/ros/resource_retriever",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "interactive_markers",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/interactive_markers",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "python_qt_binding",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/python_qt_binding",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_plot",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_plot",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ament_index",
    "org": "ament",
    "url": "https://github.com/ament/ament_index",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "performance_test",
    "org": "ros2",
    "url": "https://github.com/ros2/performance_test",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ivanpauno",
    ],
    "name": "rmw_dds_common",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_dds_common",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "depthimage_to_laserscan",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/depthimage_to_laserscan",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_topic",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_topic",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ament_package",
    "org": "ament",
    "url": "https://github.com/ament/ament_package",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "uncrustify_vendor",
    "org": "ament",
    "url": "https://github.com/ament/uncrustify_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "osrf_testing_tools_cpp",
    "org": "osrf",
    "url": "https://github.com/osrf/osrf_testing_tools_cpp",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "orocos_kinematics_dynamics",
    "org": "ros2",
    "url": "https://github.com/ros2/orocos_kinematics_dynamics",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "pybind11_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/pybind11_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "python_cmake_module",
    "org": "ros2",
    "url": "https://github.com/ros2/python_cmake_module",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_runtime_py",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_runtime_py",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "test_interface_files",
    "org": "ros2",
    "url": "https://github.com/ros2/test_interface_files",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "urdf",
    "org": "ros2",
    "url": "https://github.com/ros2/urdf",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "filters",
    "org": "ros",
    "url": "https://github.com/ros/filters",
    "validUrl": null,
  },
  {
    "maintainers": [
      "nuclearsandwich",
    ],
    "name": "ros_environment",
    "org": "ros",
    "url": "https://github.com/ros/ros_environment",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "urdfdom_headers",
    "org": "ros",
    "url": "https://github.com/ros/urdfdom_headers",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "googletest",
    "org": "ament",
    "url": "https://github.com/ament/googletest",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "google_benchmark_vendor",
    "org": "ament",
    "url": "https://github.com/ament/google_benchmark_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "audrow",
    ],
    "name": "multiarch-docker-image-generation",
    "org": "osrf",
    "url": "https://github.com/osrf/multiarch-docker-image-generation",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
      "cottsay",
    ],
    "name": "ignition_cmake2_vendor",
    "org": "ignition",
    "url": "https://github.com/ignition/ignition_cmake2_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
      "cottsay",
    ],
    "name": "ignition_math6_vendor",
    "org": "ignition",
    "url": "https://github.com/ignition/ignition_math6_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "navigation_msgs",
    "org": "ros-planning",
    "url": "https://github.com/ros-planning/navigation_msgs",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "Mimick",
    "org": "ros2",
    "url": "https://github.com/ros2/Mimick",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "ament_cmake_ros",
    "org": "ros2",
    "url": "https://github.com/ros2/ament_cmake_ros",
    "validUrl": null,
  },
  {
    "maintainers": [
      "nuclearsandwich",
    ],
    "name": "choco-packages",
    "org": "ros2",
    "url": "https://github.com/ros2/choco-packages",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "console_bridge_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/console_bridge_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "eigen3_cmake_module",
    "org": "ros2",
    "url": "https://github.com/ros2/eigen3_cmake_module",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "example_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/example_interfaces",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "mimick_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/mimick_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "performance_test_fixture",
    "org": "ros2",
    "url": "https://github.com/ros2/performance_test_fixture",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rmw_connext",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_connext",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ros2cli_common_extensions",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2cli_common_extensions",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ros_testing",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_testing",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ros_workspace",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_workspace",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_core",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_core",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_dds",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_dds",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_defaults",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_defaults",
    "validUrl": null,
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_typesupport_connext",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport_connext",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rpyutils",
    "org": "ros2",
    "url": "https://github.com/ros2/rpyutils",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
      "quarkytale",
    ],
    "name": "spdlog_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/spdlog_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "teleop_twist_joy",
    "org": "ros2",
    "url": "https://github.com/ros2/teleop_twist_joy",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "teleop_twist_keyboard",
    "org": "ros2",
    "url": "https://github.com/ros2/teleop_twist_keyboard",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tinyxml2_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/tinyxml2_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tinyxml_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/tinyxml_vendor",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "tlsf",
    "org": "ros2",
    "url": "https://github.com/ros2/tlsf",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "unique_identifier_msgs",
    "org": "ros2",
    "url": "https://github.com/ros2/unique_identifier_msgs",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "variants",
    "org": "ros2",
    "url": "https://github.com/ros2/variants",
    "validUrl": null,
  },
  {
    "maintainers": [
      "tfoote",
    ],
    "name": "rosdoc_lite",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosdoc_lite",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "laser_geometry",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/laser_geometry",
    "validUrl": null,
  },
  {
    "maintainers": [
      "scpeters",
    ],
    "name": "console_bridge",
    "org": "ros",
    "url": "https://github.com/ros/console_bridge",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "eigen_stl_containers",
    "org": "ros",
    "url": "https://github.com/ros/eigen_stl_containers",
    "validUrl": null,
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "urdf_parser_py",
    "org": "ros",
    "url": "https://github.com/ros/urdf_parser_py",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_action",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_action",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_common_plugins",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_common_plugins",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_console",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_console",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_msg",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_msg",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_publisher",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_publisher",
    "validUrl": null,
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_py_console",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_py_console",
    "validUrl": null,
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_service_caller",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_service_caller",
    "validUrl": null,
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_shell",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_shell",
    "validUrl": null,
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_srv",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_srv",
    "validUrl": null,
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tango_icons_vendor",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/tango_icons_vendor",
    "validUrl": null,
  },
]
`;

exports[`getRepos should return repos 2`] = `
[
  "wjwwood",
  "ivanpauno",
  "hidmic",
  "sloretz",
  "adityapande-1995",
  "clalancette",
  "codebot",
  "ahcorde",
  "audrow",
  "methylDragon",
  "gbiggs",
  "quarkytale",
  "mjeronimo",
  "cottsay",
  "nuclearsandwich",
  "tfoote",
  "scpeters",
]
`;
